### PR TITLE
Study Details Configuration Views

### LINKED ISSUE #1243: Move Experiment Runner config to its own view
**TL;DR**: The study config view is overwhelming.  To help fix this, move the Experiment Runner options to it's own view. 

**Narrative**
The Study edit page is large and difficult to maintain.  Moving these options it own own view will greatly improve the Study edit use experience and maintainability of code.   

**Acceptance Criteria**
A precise description of how/when this issue will be considered closed, e.g.:

- [ ] Tests
- [ ] Scoping
- [ ] Alternatives Considered
- [ ] HTML reflects changes in exp build status
- [ ] HTML reflects changes in study review status
- [ ] Modal (still) shows when user needs to fill out form.

### PR DESCRIPTION
**Pull request by okaycj**
This PR moves the experiment configuration to its own view.  

Closes #1243

Study Edit view:
Screenshot 2023-08-23 at 10 26 02 AM
Screenshot 2023-08-23 at 10 26 06 AM
Screenshot 2023-08-23 at 10 26 51 AM
Screenshot 2023-08-23 at 10 26 55 AM
Screenshot 2023-08-23 at 10 26 59 AM
Screenshot 2023-08-23 at 10 27 04 AM

Study Detail:
Screenshot 2023-08-23 at 10 27 47 AM

EFP Config:
Screenshot 2023-08-23 at 10 28 28 AM
Screenshot 2023-08-23 at 10 28 31 AM

External Config:
Screenshot 2023-08-23 at 10 32 00 AM

### CODE REVIEW

**Comment by mekline** [comment]
+1 to adding the preview button back to the study details page!

**Review by bleonar5 — APPROVED** [review]
Everything looks good to me, for what it's worth given that I can't run the code yet. The only commit that confused me was "Update version of djlint", which contained the djlint update but also a bunch of miscellaneous formatting changes.

**Comment by becky-gilbert** [comment]
@okaycj I started reviewing this and will finish ASAP. But wanted to give you a quick heads up that I've just merged my pipe death PR into `develop` (#1237), so you might get some merge conflicts when you merge this. 

Also, I noticed that we both have study migration files with the same numbered prefix (`0091`), though obviously the file names are different. Not sure the number itself matters.

**Review by becky-gilbert — APPROVED** [review]
Great work @okaycj! Looks like it was a lot of work to pull these two forms apart. Thanks for the refactoring and cleaning up too!
I tested all of the logic that we talked about (when the study should be rejected and/or require rebuilding vs not) and it all looks great to me!
I added a few comments and suggestions. And I have a few very nit-picky suggestions that I'm putting into a PR for the `exp-config` branch. Please feel free to ignore/close my PR if you don't want to include my changes. You could also ask me to remove specific commits for anything you don't want.

File: exp/views/study.py [hunk 1]
```diff
@@ -1060,3 +973,190 @@ def get_permitted_triggers(view_instance, triggers):
...
+            "scheduling": form.cleaned_data["scheduling"],
+            "other_scheduling": form.cleaned_data["other_scheduling"],
+            "study_platform": form.cleaned_data["study_platform"],
+            "other_study_platform": form.cleaned_data["other_study_platform"],
+        }
+
+        if metadata != study.metadata:
+            study.built = False
+            study.is_building = False
+            study.metadata = metadata
```
[the remark this answers is not in the document]
Comment (mekline) [reply exp/views/study.py:1160]: That's correct, if my understanding that this maintains status quo is also correct!

File: studies/templates/studies/lab_create.html [hunk 2]
```diff
@@ -16,7 +16,7 @@
         {% csrf_token %}
         {% bootstrap_form form %}
         {% form_buttons %}
-            {% bootstrap_button "Submit" %}
-        {% endform_buttons %}
-    </form>
+        {% bootstrap_button "Submit" %}
+    {% endform_buttons %}
```
Comment (becky-gilbert) [inline studies/templates/studies/lab_create.html:20]: @okaycj one other minor thing I forgot to mention: I noticed that our new linting tool isn't indenting some of these template tags properly. E.g. `form_buttons`/`endform_buttons` isn't recognized as the start/end of an indented code block here (and in several other files). Is there any way to fix this? (Obviously this shouldn't hold up merging this PR - if it's not an easy fix then I'll put it in an issue).

File: studies/forms.py [hunk 3]
```diff
@@ -432,3 +350,161 @@ class EmailParticipantsForm(forms.Form):
...
+    player_repo_url = forms.URLField(
+        label="Experiment runner code URL",
+        help_text="Leave this value alone unless you are using a custom version of the experiment builder.",
+    )
+    last_known_player_sha = forms.CharField(
+        label="Experiment runner version (commit SHA)",
+        help_text=(
+            "If you're using the default Ember Frame Player, you can see <a "
+            'href="https://github.com/lookit/ember-lookit-frameplayer/commits/master">'
+            "the commits page</a> for other commit SHA options."
```
[the remark this answers is not in the document]
Comment (okaycj) [reply studies/forms.py]: Added suggestion in commit [c892afd](https://github.com/lookit/lookit-api/pull/1244/commits/c892afdac2991232db0b707df25a37ca1ea6f749).

**Review by bleonar5 — APPROVED** [review]

File: studies/templates/studies/lab_create.html (hunk 2 above)
Comment (okaycj) [reply studies/templates/studies/lab_create.html:20]: Yeah, I noticed this as well.  It's not great and I hope they fix it in the future.  If it's a real problem, I have no problem with submitting an issue.

**Review by bleonar5 — APPROVED** [review]

**Comment by mekline** [comment]
Okay, I think I see what you mean about that copy when the first study page is visited in edit mode and the study type can no longer be changed.  Let me know if this sounds/looks more natural for both Create Study and Edit Study: 

*******
You have selected a [TYPE] study. The study type cannot be changed after creation.

Before saving for the first time, please [review the distinction between experiment builder (internal) studies and external studies](https://lookit.readthedocs.io/en/develop/index.html?highlight=external#what-is-lookit), and ask a question on [Slack](http://lookit-mit.slack.com/) if you need help selecting the right option!

**Comment by mekline** [comment]
Approved other than that copy change though!

**Comment by okaycj** [comment]
@mekline 

I am unsure about the `You have selected a [TYPE] study.` of the copy.  Currently, this text seems like its unneeded, what the user selected is just above this text.

**Pull request merged**